src/mapgen.cpp

```diff
@@ -589,13 +589,13 @@ class jmapgen_alternativly : public jmapgen_piece {
     // PieceType, they *can not* be of any other type.
     std::vector<PieceType> alternatives;
     jmapgen_alternativly() = default;
-    void apply( map &m, const size_t x, const size_t y, const float mon_density ) const override
+    void apply( map &m, const size_t x, const size_t y, const size_t xmin, const size_t ymin, const size_t xmax, const size_t ymax, const float mon_density ) const override
```

Comment:
```x``` and ```y``` are already randomized before ```apply()``` is called, but I could change where the random ```x/y``` is assigned by moving it into each ```apply``` function